File: Xamarin/Test/PhysicsTest.cs
//
// Copyright (c) 2015 Harry Cheung
//

using NUnit.Framework;
using Xamarin.Shared;

namespace Xamarin.Shared
{
	[TestFixture()]
	public class PhysicsTest
	{
		[Test()]
		public void TestDistance() {
			Assert.AreEqual(0.0, Physics.Distance(0, 0, 0));
			Assert.AreEqual(1.5, Physics.Distance(1, 1, 1));
			Assert.AreEqual(8.0, Physics.Distance(2, 2, 2));
			Assert.AreEqual(9.0, Physics.Distance(3, 0, 3));
		}

		[Test()]
		public void TestTime() {
			Assert.AreEqual(1.0, Physics.Time(1.5, 1, 1));
			Assert.AreEqual(2.0, Physics.Time(8.0, 2, 2));
			Assert.AreEqual(3.0, Physics.Time(9.0, 3, 0));
		}
	}
}

File: J2ObjC/Java/src/test/java/harrycheung/map/shared/PhysicsTest.java
//
// Copyright (c) 2015 Harry Cheung
//

package harrycheung.map.shared;

import static org.junit.Assert.assertThat;
import static org.hamcrest.CoreMatchers.is;
import harrycheung.map.shared.Physics;

import org.junit.Test;

public class PhysicsTest {

  @Test
  public void distance() {
    assertThat(Physics.distance(0, 0, 0), is(0.0));
    assertThat(Physics.distance(1, 1, 1), is(1.5));
    assertThat(Physics.distance(2, 2, 2), is(8.0));
    assertThat(Physics.distance(3, 0, 3), is(9.0));
  }

  @Test
  public void time() {
    assertThat(Physics.time(1.5, 1, 1), is(1.0));
    assertThat(Physics.time(8.0, 2, 2), is(2.0));
    assertThat(Physics.time(9.0, 3, 0), is(3.0));
  }
}
